File: benchmark/python/sieve.py
import time
limit = 10000
sieve = [True] * (limit + 1)
sieve[0] = sieve[1] = False
start = time.time() * 1000
for p in range(2, int(limit**0.5) + 1):
    if sieve[p]:
        for multiple in range(p * p, limit + 1, p):
            sieve[multiple] = False
count = 0
for x in sieve:
    if x:
        count += 1
end = time.time() * 1000
print(f"Primes up to 10000: {count}")
print(f"Time: {end - start:.0f}ms")

File: benchmark/php/sieve.php
<?php
$limit = 10000;
$sieve = array_fill(0, $limit + 1, true);
$sieve[0] = false; $sieve[1] = false;
$start = hrtime(true);
for ($p = 2; $p * $p <= $limit; $p++) {
    if ($sieve[$p]) {
        for ($m = $p * $p; $m <= $limit; $m += $p) {
            $sieve[$m] = false;
        }
    }
}
$count = 0;
for ($i = 2; $i <= $limit; $i++) {
    if ($sieve[$i]) $count++;
}
$end = hrtime(true);
$ms = ($end - $start) / 1e6;
echo "Primes up to 10000: " . $count . "
";
echo "Time: " . round($ms) . "ms
";
